packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -457,4 +474,268 @@ linter:
       printStatus('No analyzer warnings! (ran in ${elapsed}s)');
     return 0;
   }
+
+  Future<int> _analyzeWatch() async {
+    List<String> directories;
+
+    if (isFlutterRepo) {
+      printStatus('Analyzing Flutter repository:');
+      directories = <String>[];
+      directories.addAll(_gatherProjectPaths(path.absolute('examples')));
+      directories.addAll(_gatherProjectPaths(path.absolute('packages')));
+      for (String projectPath in directories)
+        printStatus('  ${path.relative(projectPath)}');
+      printStatus('');
```

Comment:
When the analysis server traverses down directories looking for dart projects (what it calls 'contexts'), it'll ignore directories named `packages`. It'll assume that they're just directories of symlinks.

I explicitly traverse into these directories looking for roots.